### DISCUSSION TITLE
need ur assistance

### DISCUSSION THREAD

**Discussion by spaakofi** [Imperial-lord/mapbox-flutter#2] [Q&A]
please i have a school project im about to work on i just dont know how to start or go about it and im suppose to use mapbox to create the app on flutter pls can u assist me

**Comment by Imperial-lord** [comment]
Hey @spaakofi. Can you tell me more about your project and what difficulty you are facing? You can include, screenshots, code or anything here.

**Reply by spaakofi** [reply]
the project im working on is to build an app which will direct people to a specific location within the country so i started using the map box navigation and also the flutter map 

I have implemented the Mapbox and other plugins to help me out but when I'm running without debugging it I always get some issues I'm trying to resolve
 
I have changed the secret token several times and I have also read the documentation  of it on MapBox and stack flow but it isn't working 

> Could not determine artifacts for com.mapbox.navigation:ui:1.2.0
      > Could not get resource '[https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/navigation/ui/1.2.0/ui-1.2.0.aar]()'.
         > Could not HEAD '[https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/navigation/ui/1.2.0/ui-1.2.0.aar]()'. Received status code 401 from server: Unauthorized

* Try:
Run with --stack trace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.



* Get more help at [https://help.gradle.org]() 
*> Could not determine artifacts for com.mapbox.navigation:ui:1.2.0
      > Could not get resource '[https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/navigation/ui/1.2.0/ui-1.2.0.aar]()'.
         > Could not HEAD '[https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/navigation/ui/1.2.0/ui-1.2.0.aar]()'. Received status code 401 from server: Unauthorized

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at [https://help.gradle.org]()

BUILD FAILED in 2s

**Reply by Imperial-lord** [reply]
Additionally, if you are still stuck, point me to the repository where you have written your code, so that I can take a look at it

**Comment by Imperial-lord** [comment]
401 unauthorised means that you have not set up the secret token properly. Have you checked the Downloads:READ option? 
Also could you tell me what device are you targeting? Is it an iOS emulator or Android emulator?

**Reply by spaakofi** [reply]
I checked all of the secret token properties but still the same option I'm getting 
I'm using my phone to do the debugging (Andriod ).

I did all the permissions which were required of me to do 

apparently, I  have started a new project trying to use your video on YouTube but still not getting some things right 
new to flutter say 2months now since I started the course. 

please help me get out with the project

**Reply by Imperial-lord** [reply]
Hey sure! Don't worry. Do you mind to put the project up in Github so that I can look at the code? I will they try running it locally and let you know where you might be going wrong.

**Reply by spaakofi** [reply]
okay sure 
i will do that right away

**Reply by Imperial-lord** [reply]
Any updates @spaakofi ?

**Reply by scoopy21** [reply]
@Imperial-lord Can u plz help me?
Although I am using your code but I cannot retrieve the 'place' value. Here is the error

[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
E/flutter (14059): #0      getParsedReverseGeocoding (package:testnavigation/helpers/mapbox_handler.dart:42:15)
E/flutter (14059): <asynchronous suspension>
E/flutter (14059): #1      _LocationFieldState._useCurrentLocationButtonHandler (package:testnavigation/widgets/location_field.dart:63:22)
E/flutter (14059): <asynchronous suspension>

the mapbox_handler code
```
// ----------------------------- Mapbox Reverse Geocoding -----------------------------
Future<Map> getParsedReverseGeocoding(LatLng latLng) async {
  var response =
  json.decode(await getReverseGeocodingGivenLatLngUsingMapbox(latLng));
  Map feature = response['features'][0];
  Map revGeocode = {
    'name': feature['text'],
    'address': feature['place_name'].split('${feature['text']}, ')[1],
    'place': feature['place_name'],
    'location': latLng
  };
  return revGeocode;
}
```

the location_field code
```
_useCurrentLocationButtonHandler() async {
    if (!widget.isDestination) {
      LatLng currentLocation = getCurrentLatLngFromSharedPrefs();

      // Get the response of reverse geocoding and do 2 things:
      // 1. Store encoded response in shared preferences
      // 2. Set the text editing controller to the address
      var response = await getParsedReverseGeocoding(currentLocation);
      sharedPreferences.setString('source', json.encode(response));
      String place = response['place'];
      widget.textEditingController.text = place;
    }
  }
```

Please help
